File: GuiaApe06/Ejercicio3.cpp
#include <iostream>   // Librería para entrada y salida de datos
#include <vector>     // Librería para usar vectores dinámicos

using namespace std;

int main() {

    // Crear un vector para almacenar productos
    vector<string> productos;

    // Variables auxiliares
    string producto, buscar;

    // Variable booleana para verificar si se encontró el producto
    bool encontrado = false;

    // Ciclo para ingresar 5 productos
    for(int i = 0; i < 5; i++) {

        // Pedir nombre del producto
        cout << "Ingrese producto: ";

        // Leer el producto completo
        getline(cin, producto);

        // Agregar el producto al vector
        productos.push_back(producto);
    }

    // Pedir el producto a buscar
    cout << "Producto a buscar: ";

    // Leer el nombre del producto a buscar
    getline(cin, buscar);

    // Recorrer el vector para buscar el producto
    for(int i = 0; i < productos.size(); i++) {

        // Comparar si el producto existe
        if(productos[i] == buscar) {

            // Cambiar el valor a verdadero si se encuentra
            encontrado = true;
        }
    }

    // Mostrar resultado de la búsqueda
    if(encontrado)

        // Mensaje si el producto existe
        cout << "Producto encontrado";

    else

        // Mensaje si no existe
        cout << "Producto no encontrado";

    return 0;
}

File: GuiaApe06/Ejercicio3.java
import java.util.ArrayList; // Librería para usar ArrayList
import java.util.Scanner;   // Librería para entrada de datos

public class Ejercicio3 {

    public static void main(String[] args) {

        // Crear objeto Scanner para leer datos
        Scanner sc = new Scanner(System.in);

        // Crear un ArrayList para guardar productos
        ArrayList<String> productos = new ArrayList<>();

        // Variables auxiliares
        String producto, buscar;

        // Ciclo para ingresar 5 productos
        for(int i = 0; i < 5; i++) {

            // Pedir nombre del producto
            System.out.print("Ingrese producto: ");

            // Leer el producto ingresado
            producto = sc.nextLine();

            // Agregar el producto al ArrayList
            productos.add(producto);
        }

        // Pedir producto a buscar
        System.out.print("Producto a buscar: ");

        // Leer el producto a buscar
        buscar = sc.nextLine();

        // Verificar si el producto existe en el ArrayList
        if(productos.contains(buscar)) {

            // Mostrar mensaje si se encuentra
            System.out.println("Producto encontrado");

        } else {

            // Mostrar mensaje si no se encuentra
            System.out.println("Producto no encontrado");
        }

        // Cerrar Scanner
        sc.close();
    }
}

